src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java

```
@@ -0,0 +1,116 @@
+            if (resultSet.next()) {
+                return Optional.of(getManufacturerFromResultSet(resultSet));
+            }
+            return Optional.empty();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't execute statement "
+                    + "to get manufacturer by id" + id, e);
+        }
+    }
+
+    @Override
+    public List<Manufacturer> getAll() {
+        List<Manufacturer> manufacturers = new ArrayList<>();
+        String selectRequest = "SELECT * FROM manufacturers WHERE is_deleted = FALSE;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement getAllManufacturersStatement
+                         = connection.prepareStatement(selectRequest)) {
+            ResultSet resultSet = getAllManufacturersStatement.executeQuery();
+            while (resultSet.next()) {
+                manufacturers.add(getManufacturerFromResultSet(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't execute statement "
+                    + "to get all manufacturers", e);
+        }
+        return manufacturers;
+    }
+
+    @Override
+    public Manufacturer update(Manufacturer manufacturer) {
+        String updateRequest = "UPDATE manufacturers"
+                + " SET name = ?, country = ? WHERE id = ? AND is_deleted = FALSE;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement updateManufacturerStatement
+                         = connection.prepareStatement(updateRequest)) {
+            updateManufacturerStatement.setString(1, manufacturer.getName());
+            updateManufacturerStatement.setString(2, manufacturer.getCountry());
+            updateManufacturerStatement.setLong(3, manufacturer.getId());
+            return updateManufacturerStatement.executeUpdate() > 0 ? manufacturer : null;
```

bad idea to return `null`, lets throw an exception instead